prestoadmin/yarn_slider/server.py

```diff
@@ -151,5 +158,97 @@ def uninstall():
     except KeyError:
         pass
 
-    local('rm %s' % (' '.join([os.path.join(SLIDER_CONFIG_DIR, f)
-                     for f in SLIDER_PKG_DEFAULT_FILES])))
+    with quiet():
+        local('rm %s' % (' '.join([os.path.join(SLIDER_CONFIG_DIR, f)
+                         for f in SLIDER_PKG_DEFAULT_FILES])))
+
+
+@contextmanager
+def remote_temp_file(local_path, remote_path):
+    result = []
+    try:
+        result = put(local_path, remote_path)
+        if len(result) == 1:
+            yield result[0]
+        else:
+            yield result
+    finally:
+        for remote_path in result:
+            run('/bin/rm "%s"' % remote_path)
+
+
+def start_app():
+    start_cmd = (
+        '%s start %s' % (get_slider_bin(env.conf), env.conf[APPNAME]))
+    with warn_only():
+        return run_slider(start_cmd, env.conf)
+
+
+BUILD = 'build'
+CREATE = 'create'
+
+
+def _build_or_create(command=BUILD):
+    appConfig = AppConfigJson()
+    resources = ResourcesJson()
+
+    with remote_temp_file(appConfig.get_config_path(), '/tmp') as \
+            remote_appConfig:
+        with remote_temp_file(resources.get_config_path(), '/tmp') as \
+                remote_resources:
+            slider_cmd = '%s %s %s --template %s --resources %s' % (
+                get_slider_bin(env.conf), command, env.conf[APPNAME],
+                remote_appConfig, remote_resources)
+            return run_slider(slider_cmd, env.conf)
+
+
+@task
+@requires_config(SliderConfig, AppConfigJson, ResourcesJson)
+@task_by_rolename(SLIDER_MASTER)
+def create():
+    return _build_or_create(command=CREATE)
+
+
+@task
+@requires_config(SliderConfig, AppConfigJson, ResourcesJson)
+@task_by_rolename(SLIDER_MASTER)
+def build():
+    return _build_or_create(command=BUILD)
+
+
+# Starting Presto doesn't actually require AppConfig and Resources, but if the
+# app doesn't exist and we have to fall back to creating it, it's too late to
+# prompt the user for anything we don't have a default for when we call
+# create().
+@task
+@requires_config(SliderConfig, AppConfigJson, ResourcesJson)
+@task_by_rolename(SLIDER_MASTER)
+def start():
+    appname = env.conf[APPNAME]
+    start_result = start_app()
+    if start_result.return_code == EXIT_SUCCESS:
```

Comment:
I think you'll get useful output from the call to slider create. We aren't hiding it from the user.